app/src/main/java/com/example/plantsapp/presentation/ui/plantdetail/PlantDetailFragment.kt

```diff
@@ -48,6 +50,13 @@ class PlantDetailFragment : Fragment(R.layout.fragment_plant_detail) {
                 activity?.title = it
             }
 
+            isLoading.observe(viewLifecycleOwner) { isLoading ->
+                when (isLoading) {
+                    true -> loadingDialog.show()
+                    false -> loadingDialog.dismiss()
+                }
+            }
```

Comment:
How about move this thing to LoadingDialog file as an Extension? 
It will look like
fun LoadingDialog.connectWith(isLoadingLiveData: LiveData, lifecycleOwner: LifecycleOwner)